### PR TITLE
algorithm: Add juggler sequence

### PR DESCRIPTION
**Pull request by itsAkshayDubey**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/juggler_sequence.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        raise TypeError(f"Input value of [number={number}] must be greater than 0")
+    sequence = [number]
+    while number != 1:
+        if number % 2 == 0:
+            temp = int(math.floor(math.sqrt(number)))
+        else:
+            temp = int(
+                math.floor(math.sqrt(number) * math.sqrt(number) * math.sqrt(number))
+            )
+        number = temp
```
Comment (cclauss) [inline maths/juggler_sequence.py:53]: `temp` is not needed...
```suggestion
            number = int(math.floor(math.sqrt(number)))
        else:
            number = int(
                math.floor(math.sqrt(number) * math.sqrt(number) * math.sqrt(number))
            )
```

Comment (itsAkshayDubey) [reply maths/juggler_sequence.py]: Fix in [this](https://github.com/TheAlgorithms/Python/pull/7985/commits/f67bf3e910837091ef13f48a1ce5ff293309cc64) commit. Kindly have a look.

Thanks

File: maths/juggler_sequence.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    if number < 1:
+        raise ValueError(f"Input value of [number={number}] must be a positive integer")
+    sequence = [number]
+    while number != 1:
+        if number % 2 == 0:
+            number = int(math.floor(math.sqrt(number)))
+        else:
+            number = int(
+                math.floor(math.sqrt(number) * math.sqrt(number) * math.sqrt(number))
+            )
```
Comment (CaedenPH) [inline maths/juggler_sequence.py:49]:
```suggestion
            number = math.floor(math.sqrt(number) * math.sqrt(number) * math.sqrt(number))
```

File: maths/juggler_sequence.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    ValueError: Input value of [number=-1] must be a positive integer
+    """
+    if not isinstance(number, int):
+        raise TypeError(f"Input value of [number={number}] must be an integer")
+    if number < 1:
+        raise ValueError(f"Input value of [number={number}] must be a positive integer")
+    sequence = [number]
+    while number != 1:
+        if number % 2 == 0:
+            number = int(math.floor(math.sqrt(number)))
```
Comment (CaedenPH) [inline maths/juggler_sequence.py:45]:
```suggestion
            number = math.floor(math.sqrt(number))
```
`math.floor` returns `int`

**Review by CaedenPH — COMMENTED** [review]
Looks good 
@cclauss Can you review?

**Comment by cclauss** [comment]
Can we have tests for larger numbers like 10 and 25?

File: maths/juggler_sequence.py (hunk 3 above)
Comment (itsAkshayDubey) [reply maths/juggler_sequence.py]: Fixed in [this](https://github.com/TheAlgorithms/Python/pull/7985/commits/dc5db724bd939e08a44ae13fa9030be0734b8582) commit. Kindly have a look.

**Comment by itsAkshayDubey** [comment]
> Can we have tests for larger numbers like 10 and 25?

I have added tests for 10 and 25 in [this](https://github.com/TheAlgorithms/Python/pull/7985/commits/20da7a3d70f10535153cbcba6bb2f1d8088a69ad) commit, kindly have a look.

**Pull request merged**